nautobot/utilities/views.py

```diff
@@ -151,3 +186,1062 @@ def get_return_url(self, request, obj=None):
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
+
+        try:
+            select_template([f"{app_label}/{model_opts.model_name}_{view_type}.html"])
+            return f"{app_label}/{model_opts.model_name}_{view_type}.html"
+        except TemplateDoesNotExist:
+            return f"generic/object_{view_type}.html"
+
+
+class ObjectDetailViewMixin(NautobotViewSetMixin):
+    """
+    Retrieve a single object for display.
+    queryset: The base queryset for retrieving the object
+    template_name: Name of the template to use
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.routes = self.define_routes()
+
+    def define_routes(self):
+        return super().define_routes() + [
+            Route(
+                url=r"^{prefix}/{lookup}/$",
+                mapping={
+                    "get": "handle_object_detail_get",
+                },
+                name="{basename}",
+                detail=True,
+                initkwargs={"suffix": "Detail"},
+            ),
+        ]
+
+    def get_changelog_url(self, instance):
+        """Return the changelog URL for a given instance."""
+        meta = self.queryset.model._meta
+
+        # Don't try to generate a changelog_url for an ObjectChange.
+        if meta.model_name == "objectchange":
+            return None
+
+        route = f"{meta.app_label}:{meta.model_name}_changelog"
+        if meta.app_label in settings.PLUGINS:
+            route = f"plugins:{route}"
+
+        # Iterate the pk-like fields and try to get a URL, or return None.
+        fields = ["pk", "slug"]
+        for field in fields:
+            if not hasattr(instance, field):
+                continue
+
+            try:
+                return reverse(route, kwargs={field: getattr(instance, field)})
+            except NoReverseMatch:
+                continue
+
+        # This object likely doesn't have a changelog route defined.
+        return None
+
+    def handle_object_detail_get(self, request, *args, **kwargs):
+        """
+        Generic GET handler for accessing an object by PK or slug
+        """
+        instance = get_object_or_404(self.queryset, **kwargs)
+        self.context = self.get_extra_context(request, "detail", instance)
+        return render(
+            request,
+            self.template_name,
+            {
+                "object": instance,
+                "verbose_name": self.queryset.model._meta.verbose_name,
+                "verbose_name_plural": self.queryset.model._meta.verbose_name_plural,
+                **self.context,
+                "changelog_url": self.get_changelog_url(instance),
+            },
+        )
+
+
+class ObjectListViewMixin(NautobotViewSetMixin):
```

Comment:
Not sure if this was originally based on older code, but it's missing some of the functionality present in `generic.ObjectListView`, particularly the `get_filter_params()` and `validate_action_buttons()` methods and their usage in the `get`/`handle_object_list_get` method.

Reply:
I saw and added `validate_action_buttons()` but I dont think I can quite locate `get_filter_params()` can you point to it?

Reply:
https://github.com/nautobot/nautobot/blob/next/nautobot/core/views/generic.py#L158, https://github.com/nautobot/nautobot/blob/next/nautobot/core/views/generic.py#L226